mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDescriptorDaoImpl.java

```diff
@@ -118,16 +118,18 @@ private String getQueryByEventTypes(List<String> eventTypes) {
   }
 
   @Override
-  public Future<Boolean> delete(String eventType) {
+  public Future<Void> delete(String eventType) {
     Promise<RowSet<Row>> promise = Promise.promise();
-    try {
-      String query = format(DELETE_BY_ID_SQL, MODULE_SCHEMA, TABLE_NAME);
-      pgClientFactory.getInstance().execute(query, Tuple.of(eventType), promise);
-    } catch (Exception e) {
-      LOGGER.error("Error deleting EventDescriptor with event type '{}'", eventType, e);
-      promise.fail(e);
-    }
-    return promise.future().map(updateResult -> updateResult.rowCount() == 1);
+    String query = format(DELETE_BY_ID_SQL, MODULE_SCHEMA, TABLE_NAME);
+    pgClientFactory.getInstance().execute(query, Tuple.of(eventType), promise);
+
+    return promise.future().flatMap(updatedRow -> {
+      if(updatedRow.rowCount() == 1) {
+        return Future.succeededFuture();
+      }
+      return Future.failedFuture(new NotFoundException(
```

Comment:
We're failing with NotFoundException, but is it always "not found"? What if 2 rows were deleted?

Reply:
Interesting... before it was used to check for updateRow == 1, so I have tried to keep consistent with previous case